android/src/main/kotlin/com/adyen/checkout/flutter/components/ComponentWrapperView.kt

```diff
@@ -19,60 +19,64 @@ import kotlinx.coroutines.launch
-        this.activity = activity
-        this.componentFlutterApi = componentFlutterApi
-    }
+    @JvmOverloads
+    constructor(
+        context: Context,
+        attrs: AttributeSet? = null,
+        defStyleAttr: Int = 0
+    ) : ConstraintLayout(context, attrs, defStyleAttr) {
+        constructor(
+            activity: ComponentActivity,
+            componentFlutterApi: ComponentFlutterInterface
+        ) : this(context = activity) {
+            this.activity = activity
+            this.componentFlutterApi = componentFlutterApi
+        }
 
-    private lateinit var activity: ComponentActivity
-    private lateinit var componentFlutterApi: ComponentFlutterInterface
-    private val screenDensity = context.resources.displayMetrics.density
+        private lateinit var activity: ComponentActivity
+        private lateinit var componentFlutterApi: ComponentFlutterInterface
+        private val screenDensity = context.resources.displayMetrics.density
 
-    init {
-        inflate(context, R.layout.component_wrapper_view, this)
-    }
-
-    fun <T> addComponent(cardComponent: T) where T : ViewableComponent, T : Component {
-        with(findViewById<AdyenComponentView>(R.id.adyen_component_view)) {
-            attach(cardComponent, activity)
-            addComponentHeightObserver()
+        init {
+            inflate(context, R.layout.component_wrapper_view, this)
         }
-    }
 
-    private fun addComponentHeightObserver() {
-        ComponentHeightMessenger.instance().removeObservers(activity)
-        ComponentHeightMessenger.instance().observe(activity) {
-            activity.lifecycleScope.launch {
-                //We need to wait for animation to finish e.g. when scheme icons disappear
-                delay(300)
-                updateComponentViewHeight()
+        fun <T> addComponent(cardComponent: T) where T : ViewableComponent, T : Component {
+            with(findViewById<AdyenComponentView>(R.id.adyen_component_view)) {
+                attach(cardComponent, activity)
+                addComponentHeightObserver()
             }
         }
-    }
 
-    private fun updateComponentViewHeight() {
-        val cardViewHeight = findViewById<FrameLayout>(R.id.frameLayout_componentContainer)?.getChildAt(0)?.height
-        if (cardViewHeight == null) {
-            activity.lifecycleScope.launch() {
-                //View not rendered therefore we try again after delay.
-                //This is a workaround because there is currently no notifier from the native view.
-                delay(100)
-                updateComponentViewHeight()
+        private fun addComponentHeightObserver() {
+            ComponentHeightMessenger.instance().removeObservers(activity)
+            ComponentHeightMessenger.instance().observe(activity) {
+                activity.lifecycleScope.launch {
+                    // We need to wait for animation to finish e.g. when scheme icons disappear
+                    delay(300)
```

Comment:
Instead of having a hardcoded value you could use `context.resources.getInteger(android.R.integer.config_mediumAnimTime)` or `config_shortAnimTime` if that fits better. This way it follows the system's animation speed